streams/src/main/java/org/apache/kafka/streams/state/internals/ThreadCache.java

```diff
@@ -158,15 +169,24 @@ public void put(final String namespace, final Bytes key, final LRUCacheEntry val
         numPuts++;
 
         final NamedCache cache = getOrCreateCache(namespace);
-        cache.put(key, value);
-        maybeEvict(namespace);
+
+        synchronized (cache) {
+            final long oldSize = cache.sizeInBytes();
+            cache.put(key, value);
```

Comment:
ditto here as well -- just return the change in bytes after the put? 

same for all applicable cases, I'll stop commenting on every one but you get the gist